### PR TITLE
Add automation component and documentation.

### PR DESCRIPTION
**Pull request by TechPreacher**
- IotTelemetrySimulator.Atuomation : The .NET Standards 2.1 automation library.
  - appsettings.json : Configuration file.
- IotTelemetrySimulator.Atuomation.Console : A .NET Core 3.1 console app that illustrates how to use the automation component.
- automation.md : Documentation of the automation component.

### CODE REVIEW

**Comment by TechPreacher** [comment]
I can't add reviewers as I am not a contributor, so I am asking you this way, @fbeltrao :)

File: README.md [hunk 1]
```diff
@@ -102,13 +102,13 @@ Output:
 
 Running with docker:
 
-```
+```powershell
```
Comment (fbeltrao) [inline README.md:105]: I thinks this one is not PowerShell friendly. It is for cmd/bash

File: automation.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
+# Automation
+
+## Overview
+
+The `IotTelemtrySimulator.Automation` .NET Standards 2.1 library allows you to run the IoT Telemetry Simulator as part of a pipeline or any other automation framework.
```
Comment (fbeltrao) [inline automation.md]: nit: .NET Standard 2.1

File: automation.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+It authenticates with Azure either using managed identities or a service principal, sets up the Azure resource group, container group(s) and container instance(s) needed, pulls the IotTelemetrySimulator Docker container from an Azure container registry, configures and runs the simulator, waits for it to finish, pulls and outputs it's log files and then tears down the Container instances and groups and the resource group.
+
+![IotTelemetrySimulator Automation Architecture](/docs/images/iottelemetrysimulator_automation_architecture.png)
+
+This way, automated load tests can be run as part of a CD/CI pipeline or any automation framework that allows executing arbitrary code.
+
+## Project Structure
+
+[IotTelemetrySimulator.Atuomation](/src/IotTelemetrySimulator.Automation) : The .NET Standards 2.1 automation library.
```
Comment (fbeltrao) [inline automation.md]: nit: .NET Standard 2.1

File: src/IotTelemetrySimulator.Automation/ExceptionHandler.cs [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+                            return true;
+                        }
+                    }
+
+                    break;
+                }
+                else if (exception is NullReferenceException)
+                {
+                    break;
+                }
```
Comment (fbeltrao) [inline src/IotTelemetrySimulator.Automation/ExceptionHandler.cs]: nit: this else if is not needed

File: automation.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+If you chose *Service Principal* authentication, you need to create an Active Directory application with the rights to create and remove resources in the desired Azure subscription. To achieve this, create a new application in Azure Active Directory and then navigate to "Subscriptions &rarr; [Subscription Id] &rarr; Access Control (IAM) &rarr; Role Assignments" and then add the newly created AAD Application with "Contributor" role.
+
+If you chose *Managed Identity* authentication, the application will run with the permissions of the currently logged in user or the Azure DevOps pipeline's permissions. Make sure that you are logged in to the proper Azure account using the [AZ command line interface tool](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest).
+
+## Configuration
+
+The configuration for the automation component is kept in [appsettings.json](/src/IotTelemetrySimulator.Automation/appsettings.json) or for local development purposes in **appsettings.Development.json**. It contains the following fields
+
+|Parameter|Description|Remarks
```
Comment (fbeltrao) [inline automation.md]: The table in GH is cut. Wonder if adding breaking lines to some of the 2nd column items can improve it.

File: automation.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+|AzureSubscriptionId|Azure Subscription Id|Only used with Managed Identity authentication method|
+|AzureRegion|Azure Region as formatted in the [Microsoft.Azure.Management.ResourceManager.Fluent.Core.Region Class' Fields](https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.management.resourcemanager.fluent.core.region?view=azure-dotnet)|All resources will be created in this region.|
+|ResourceGroupName|The name of the Azure Resource Group created by to host the Container Groups|
+|ContainerGroupName|The name prefix under which the Azure Container Groups will be created|
+|ContainerImage|The full path to the docker image to be pulled by the Azure Container Instances. For example: `acrname.azurecr.io/azureiot-telemetrysimulator:latest`|The automation component has been configured to pull the image from an Azure Container Registry.
+|AcrServer|The name of the Azure Container Registry. I.e. `acrname.azurecr.io`|
+|AcrUsername|The name of user with which to access the Azure Container Registry|
+|AcrPassword|The above user's Azure Container Registry password|
```
Comment (fbeltrao) [inline automation.md]: Can I still use the image from docker hub (fbeltrao/xxx)?
Having to create an own image on ACR might be useful in some projects, but it is an additional step that might not be required.

Comment (TechPreacher) [reply automation.md]: I'll add the feature. It's a bit of work.

File: src/IotTelemetrySimulator.Automation/ExceptionHandler.cs (hunk 4 above)
Comment (TechPreacher) [reply src/IotTelemetrySimulator.Automation/ExceptionHandler.cs]: Remove these lines entirely?

File: README.md (hunk 1 above)
Comment (TechPreacher) [reply README.md:105]: Oops. Fixed.

File: src/IotTelemetrySimulator.Automation/ExceptionHandler.cs (hunk 4 above)
Comment (fbeltrao) [reply src/IotTelemetrySimulator.Automation/ExceptionHandler.cs]: Yep
```c#
                else if (exception is NullReferenceException)
                {
                    break;
                }
```

File: automation.md (hunk 5 above)
Comment (TechPreacher) [reply automation.md]: Changed to 2 columns. I couldn't get it to work otherwise.

**Review by fbeltrao — APPROVED** [review]

**Pull request merged**